### PR TITLE
[Instrumentation.AspNetCore] Fix span names when http.request.method is reporter as _OTHER

### LINKED ISSUE #5461: [Instruementation.AspNetCore] Fix span names according to sem conv 1.24.0
@Kielek - you mind fixing this in [OpenTelemetry.Instrumentation.AspNetCore](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Instrumentation.AspNetCore) as well?

_Originally posted by @vishweshbankwar in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1607#discussion_r1531242102_

### PR DESCRIPTION
**Pull request by Kielek**
Fixes #5461
Design discussion issue #

## Changes

* Fixed an issue for span names when `http.request.method` is reporter as `_OTHER`.
  Now, span names starts with `HTTP`, previously it was started by `_OTHER`.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* ~~[ ] Changes in public API reviewed (if applicable)~~

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,10 @@
   the original value `Get` will be stored in `http.request.method_original`.
   ([#5471](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5471))
 
+* Fixed an issue for span names when `http.request.method` is reporter as `_OTHER`.
+  Now, span names starts with `HTTP`, previously it was started by `_OTHER`.
+  ([#5484](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5484))
```
Comment (pellared) [inline src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:18]:
```suggestion
* Fixed the name of spans that have `http.request.method` attribute set to `_OTHER`.
  Now, the span name have `HTTP` prefix, previously it was `_OTHER`.
  ([#5484](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5484))
```

**Review by pellared — APPROVED** [review]

File: test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs [hunk 2]
```diff
@@ -629,18 +629,18 @@ void ConfigureTestServices(IServiceCollection services)
...
+    [InlineData("GET", "GET", null, "GET")]
+    [InlineData("PUT", "PUT", null, "PUT")]
+    [InlineData("HEAD", "HEAD", null, "HEAD")]
+    [InlineData("OPTIONS", "OPTIONS", null, "OPTIONS")]
+    [InlineData("PATCH", "PATCH", null, "PATCH")]
+    [InlineData("Get", "GET", "Get", "GET")]
+    [InlineData("POST", "POST", null, "POST")]
+    [InlineData("TRACE", "TRACE", null, "TRACE")]
+    [InlineData("CUSTOM", "_OTHER", "CUSTOM", "HTTP")]
+    public async Task HttpRequestMethodIsSetAsPerSpec(string originalMethod, string expectedMethod, string expectedOriginalMethod, string expectedDisplayName)
```
Comment (vishweshbankwar) [inline test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs:643]:
```suggestion
    public async Task HttpRequestMethodAndActivityDisplayIsSetAsPerSpec(string originalMethod, string expectedMethod, string expectedOriginalMethod, string expectedDisplayName)
```

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,10 @@
   the original value `Get` will be stored in `http.request.method_original`.
   ([#5471](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5471))
 
+* Fixed the name of spans that have `http.request.method` attribute set to `_OTHER`.
+  Now, the span name have `HTTP` prefix, previously it was `_OTHER`.
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:17]:
```suggestion
  The span name will be set as `HTTP {http.route}` as per the [specification](https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/http/http-spans.md#name).
```

Comment (vishweshbankwar) [reply src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:17]: Need the changelog entry for HttpClients too.

Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:17]: No behavioral changes in HTTP client. Only common method is used.

File: src/Shared/RequestMethodHelper.cs [hunk 4]
```diff
@@ -62,9 +64,11 @@ public static void SetHttpMethodTag(Activity activity, string originalHttpMethod
...
 
-    public static void SetHttpClientActivityDisplayName(Activity activity, string method)
+    public static void SetActivityDisplayName(Activity activity, string method, string? httpRoute = null)
     {
-        // https://github.com/open-telemetry/semantic-conventions/blob/v1.23.0/docs/http/http-spans.md#name
-        activity.DisplayName = KnownMethods.TryGetValue(method, out var httpMethod) ? httpMethod : "HTTP";
+        // https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/http/http-spans.md#name
+
+        var namePrefix = KnownMethods.TryGetValue(method, out var httpMethod) ? httpMethod : "HTTP";
+        activity.DisplayName = string.IsNullOrEmpty(httpRoute) ? namePrefix : $"{namePrefix} {httpRoute}";
```
[the remark this answers is not in the document]
Comment (vishweshbankwar) [reply src/Shared/RequestMethodHelper.cs:72]: I think this is good thing to consider but outside of this PR.

[the remark this answers is not in the document]
Comment (Kielek) [reply src/Shared/RequestMethodHelper.cs:72]: Separate issue created: #5488

**Pull request merged**